### PR TITLE
fix UMLS login header issue

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE PR

Jira Ticket: [MAT-4453](https://jira.cms.gov/browse/MAT-4453)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included
* [x] No extemporaneous files are included (i.e Complied files or testing results)
* [x] This PR is into the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [x] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/components/MainNavBar/MainNavBar.tsx [hunk 1]
```diff
@@ -19,13 +19,17 @@ const MainNavBar = () => {
...
+      terminologyServiceApi.checkLogin().then((value) => {
         setIsLoggedInToUMLS(true);
       })
-      .catch((err) => {});
+    );
+    return loginStatus;
+  };
+
+  useEffect(() => {
+    checkLoginToUMLS().catch((err) => {});
```
Comment (sb-prateekkeerthi) [inline src/components/MainNavBar/MainNavBar.tsx]: shouldn't we display the thrown errors to the users ?

File: src/components/MainNavBar/MainNavBar.tsx [hunk 2]
```diff
@@ -40,6 +44,9 @@ const MainNavBar = () => {
   };
 
   const { authState } = useOktaAuth();
+  if (authState?.isAuthenticated) {
+    checkLoginToUMLS().catch((err) => {});
```
Comment (sb-prateekkeerthi) [inline src/components/MainNavBar/MainNavBar.tsx]: shouldn't we display the thrown errors to the users ?

**Review by sb-prateekkeerthi — CHANGES_REQUESTED** [review]

File: src/components/MainNavBar/MainNavBar.tsx (hunk 2 above)
Comment (sb-cecilialiu) [reply src/components/MainNavBar/MainNavBar.tsx]: @sb-prateekkeerthi Thanks a lot for your quick review! I personally think it's not necessary, as the api call errors will be shown anyway, so there is no need to do it twice. Same as the above comment.

Comment (sb-prateekkeerthi) [reply src/components/MainNavBar/MainNavBar.tsx]: @sb-cecilialiu  I see that checkLoginToUMLs(in madie-utils) already has a catch block then I think you need not mention the catch block here again

Comment (sb-cecilialiu) [reply src/components/MainNavBar/MainNavBar.tsx]: @sb-prateekkeerthi In madie-utils it throws the error.

File: src/components/MainNavBar/MainNavBar.tsx [hunk 3]
```diff
@@ -17,16 +17,22 @@ const MainNavBar = () => {
...
       })
-      .catch((err) => {});
-  }, []);
+    );
+    return loginStatus;
+  };
+  const { authState } = useOktaAuth();
+  useEffect(() => {
+    if (authState?.isAuthenticated && !isLoggedInToUMLS) {
+      checkLoginToUMLS().catch((err) => {});
```
Comment (sb-prateekkeerthi) [inline src/components/MainNavBar/MainNavBar.tsx]: I think you can directly write the checkLoginToUMLS code directly in this useEffect , this kinda removes the duplicate code

**Review by sb-prateekkeerthi — CHANGES_REQUESTED** [review]

File: src/components/MainNavBar/MainNavBar.tsx (hunk 3 above)
Comment (sb-cecilialiu) [reply src/components/MainNavBar/MainNavBar.tsx]: This is done, thanks so much Prateek!!!

**Review by sb-prateekkeerthi — APPROVED** [review]
